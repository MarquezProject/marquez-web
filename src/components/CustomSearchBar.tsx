import React, { FunctionComponent, useState } from 'react'
import SearchBar from 'material-ui-search-bar'
import { findMatchingEntities } from '../actionCreators'
import {
  withStyles,
  createStyles,
  WithStyles as IWithStyles,
  Theme as ITheme
} from '@material-ui/core/styles'

import { useHistory } from "react-router-dom";

interface IProps {
  findMatchingEntities: typeof findMatchingEntities
  setShowJobs: (bool: boolean) => void
  showJobs: boolean
}

type IAllProps = IWithStyles<typeof styles> & IProps

const styles = (_theme: ITheme) => {
  return createStyles({
    search: {
      position: 'fixed',
      bottom: '52vh',
      width: '90%',
      left: '5%',
      zIndex: 4
    }
  })
}

const CustomSearchBar: FunctionComponent<IAllProps> = props => {
  const [search, setSearch] = useState("")
  const { classes } = props
  let history = useHistory();

  const searchChanged = (searchString: string) => {
    setSearch(searchString)
    props.findMatchingEntities(searchString)
    searchString == '' ? props.setShowJobs(false) : props.setShowJobs(true)
  }

  const cancelledSearch = () => {
    setSearch('')
    searchChanged('')
  }

  const onRequestSearch = () => {
    history.push("/");
  }

  return (
    <SearchBar
      className={classes.search}
      value={search}
      onChange={searchChanged}
      onCancelSearch={cancelledSearch}
      onRequestSearch={onRequestSearch}
    />
  )
}


export default withStyles(styles)(CustomSearchBar)
