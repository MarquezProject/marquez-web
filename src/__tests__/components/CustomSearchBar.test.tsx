import * as React from 'react'
import { shallow } from 'enzyme'

import CustomSearchBar from '../../components/CustomSearchBar'

describe('CustomSearchBar Component', () => {
const wrapper = shallow(<CustomSearchBar />)
  it('Should render', () => {
    expect(wrapper.exists()).toBe(true)
  })
  it('renders a snapshot that matches previous', () => {
    expect(wrapper).toMatchSnapshot()
  })

  // TODO figure out how to set state of search bar
  // wrapper.setState({search: 'test'})

  // it('should render text explaning that there was no matching dataset found', () => {
  //   expect(
  //     wrapper
  //       .find(Typography)
  //       .text()
  //       .toLowerCase()
  //   ).toContain('test')
  // })
})
