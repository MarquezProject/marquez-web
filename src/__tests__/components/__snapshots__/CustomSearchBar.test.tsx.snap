// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomSearchBar Component renders a snapshot that matches previous 1`] = `
<CustomSearchBar
  classes={
    Object {
      "search": "CustomSearchBar-search-1",
    }
  }
/>
`;
